src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -368,12 +368,30 @@ protected virtual SamlConditions CreateConditions(SecurityTokenDescriptor tokenD
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(new Uri(tokenDescriptor.Audience)));
+            var uriList = createUriList(tokenDescriptor);
+            if (!uriList.IsNullOrEmpty())
+                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(uriList));
 
             return conditions;
         }
 
+        private static List<Uri> createUriList(SecurityTokenDescriptor tokenDescriptor)
+        {
+            var uriList = new List<Uri>();
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
+            {
+                foreach (var audience in tokenDescriptor.Audiences.Where(aud => !string.IsNullOrWhiteSpace(aud)))
```

Comment:
Let's avoid using LINQ, where in this case on a perf sensitive path.

Reply:
removed